backend/services/syson-services/src/main/java/org/eclipse/syson/services/UtilService.java

```diff
@@ -199,7 +200,10 @@ public <T extends Element> T findByNameAndType(Collection<EObject> roots, String
      */
     private <T extends Element> T findInRootByQualifiedNameAndTypeFrom(EObject root, List<String> qualifiedName, Class<T> elementType) {
         T element = null;
-        if (root instanceof Element rootElt && this.nameMatches(rootElt, qualifiedName.get(0))) {
+        if (root instanceof Namespace namespace && namespace.eContainer() == null && namespace.getName() == null) {
+            // Ignore top-level namespaces with no name, they aren't part of the qualified name
+            element = this.findByQualifiedNameAndTypeFrom(namespace, qualifiedName, elementType);
```

Comment:
Fixes a bug introduced with top-level Namespaces. According to the specification (7.2.5.3), the name of a top-level element is not qualified by the name of its containing root namespace. The existing implementation was matching the first segment of the qualified name with the name of the root namespace, which is incorrect.

Reply:
Are you sure about that fix? It seems you only call the same method with the same parameters.

Reply:
I am not sure I understand. The method calls `findByQualifiedNameAndTypeFrom`, it doesn't call itself. 
The goal was to skip the call to `this.nameMatches(rootElt, qualifiedName.get(0)` if the element is a top-level, unnamed namespace, and start the search in the children of the namespace instead. 

Another option was to change the implementation of `getAllRootsInResourceSet`, but I don't think it's a good idea because namespace are still roots even if they aren't used in qualified name resolution.